operator/cmd/mesh/profile-dump.go

```diff
@@ -150,7 +160,80 @@ func profileDump(args []string, rootArgs *rootArgs, pdArgs *profileDumpArgs, l c
 		l.Print(j + "\n")
 	case yamlOutput:
 		l.Print(y + "\n")
+	case flagsOutput:
+		f, err := yamlToFlags(y)
+		if err != nil {
+			return err
+		}
+		l.Print(strings.Join(f, "\n") + "\n")
 	}
 
 	return nil
 }
+
+// Convert the generated YAML to --set flags
+func yamlToFlags(yml string) ([]string, error) {
+	// YAML objects are not completely compatible with JSON
+	// objects. Let yaml.YAMLToJSON handle the edge cases and
+	// we'll re-encode the result to pretty JSON.
+	uglyJSON, err := yaml.YAMLToJSON([]byte(yml))
+	if err != nil {
+		return []string{}, err
+	}
+	var decoded map[string]interface{}
+	if err := json.Unmarshal(uglyJSON, &decoded); err != nil {
+		return []string{}, err
+	}
+	spec, ok := decoded["spec"]
+	if !ok {
+		// Fall back to showing the entire spec.
+		// (When --config-path is used there will be no spec to remove)
+		spec = decoded
+	}
+	setflags, err := walk("", "", spec)
+	if err != nil {
+		return []string{}, err
+	}
+	sort.Strings(setflags)
+	return setflags, nil
+}
+
+func walk(path, separator string, obj interface{}) ([]string, error) {
+	if _, ok := pathBlacklist[path]; ok {
+		return []string{}, nil
+	}
+
+	switch v := obj.(type) {
+	case map[string]interface{}:
```

Comment:
doh! another tree traversal :-) i understand why it's needed here but the case code duplication bugs me. it would be great to write this using reflect rather than type assertions. i think most of the tools are already there in util/reflect.go. i plan to take care of this in 1.7 eventually so feel free to ignore this comment if doing it doesn't appeal to you.

Reply:
I prefer Go type switching over reflection.  I did a search for "visitor" in the _operator_ directory and decided there was no existing visitor pattern for IOP.  If there is one, I would prefer to use a visitor pattern to rolling my own -- point me to it!